elements/lisk-chain/src/block_reward.ts

```diff
@@ -158,13 +158,18 @@ export const applyFeeAndRewards = async (
 	// This is necessary only for genesis block case, where total fee is 0, which is invalid
 	// Also, genesis block cannot be reverted
 	generator.balance += givenFee > 0 ? givenFee : BigInt(0);
-	const totalFeeBurntStr = await stateStore.chain.get(CHAIN_STATE_BURNT_FEE);
-	let totalFeeBurnt = BigInt(totalFeeBurntStr ?? 0);
+	const totalFeeBurntBuffer = await stateStore.chain.get(CHAIN_STATE_BURNT_FEE);
+	let totalFeeBurnt = BigInt(
+		totalFeeBurntBuffer ? JSON.parse(totalFeeBurntBuffer.toString('utf-8')) : 0,
```

Comment:
I am not sure if `utf-8` is the correct value, as the documentation suggests `utf8` always. 

https://nodejs.org/api/buffer.html#buffer_buf_tostring_encoding_start_end

```suggestion
		totalFeeBurntBuffer ? JSON.parse(totalFeeBurntBuffer.toString('utf8')) : 0,
```